src/main/java/com/kodilla/ecommercee/domain/Product.java

```diff
@@ -29,15 +29,16 @@
     @NotNull
     @Column(name = "GROUP_ID")
     private String groupId;
-//    @ManyToMany(cascade = CascadeType.ALL)
-//    @JoinTable(
-//            name="JOIN_PRODUCTS_CARTS",
-//            joinColumns = {@JoinColumn(name = "CART_ID",
-//                    referencedColumnName = "CART_ID")},
-//            inverseJoinColumns = {@JoinColumn(name = "PRODUCT_ID",
-//                    referencedColumnName ="PRODUCT_ID")}
-//    )
-//    private List<Cart> cartList = new ArrayList<>();
+    @ManyToMany(cascade = CascadeType.ALL)
+    @JoinTable(
+            name="JOIN_PRODUCTS_CARTS",
+            joinColumns = {@JoinColumn(name = "PRODUCT_ID",
+                    referencedColumnName ="PRODUCT_ID")},
+            inverseJoinColumns = {@JoinColumn(name = "CART_ID",
+                    referencedColumnName = "CART_ID")}
+    )
+    private List<Cart> cartList = new ArrayList<>();
 
+//@ManyToMany(cascade = CascadeType.ALL, mappedBy = "productList")
```

Comment:
Trzeba tu będzie chyba wybrać po której stronie relacji będzie @JoinTable a po której mappedBy. :)
Jako że mam encję Cart to u siebie zrobiłem @JoinTable a w encji Product (która jest Twoim taskiem) właśnie mappedBy. Zrobiłem z tego pull requesta. Wprowadziłem też małe zmiany w moim branchu w encji Product: zmieniłem typ danych na Double przy polu price (żeby pasowało do zamodelowanego schematu tablicy, tam jest bigdecimal) oraz zmieniłem nazwę kolumny identyfikacyjnej z PRODUCT_ID na ID (tak jak w schemacie). W sumie to nie wiem czy powinienem robić takie zmiany w encji która jest Twoim taskiem......

Reply:
ok może zostać po Twojej @JoinTable a u mnie mappedBy. Tylko nie bardzo wiem jak mam zobaczyć Twoje zmiany w mojej encji - zaczekać aż Twój pull request zostanie zaaprobowany czy zrobić je też u siebie teraz?

Reply:
Hmm ale po zrobieniu tych zmian projekt się nie buduje...com.kodilla.ecommercee.EcommerceeApplicationTests > contextLoads FAILED https://prnt.sc/12j1b41 wiesz może co tu jest nie tak?

Reply:
sam się nad tym głowię jak to łączyć :D liczę, że jutrzejsze spotkanie z Andrzejem coś rozjaśni :) zrobiłem pull requesta, więc dopóki on nie będzie zaakceptowany to chyba nie da się tego podejrzeć u siebie... pewnie są jakieś magiczne (albo nie :P) komendy gita na tę okazję, ale tak jak napisałem w daily, wciąż walczę z tym gitem

Reply:
To może jednak zostawię u siebie ten bardziej rozbudowany opis relacji (bo wtedy projekt się  buduje) a Ty zrób w cart mappedBy productList ok?

Reply:
chyba powinno być mappedBy = "listOfProducts" w linii 40
encję Cart masz taką jak ja w ostatnim pull requescie?

Reply:
Na razie widzę taką encję carts @Data
@Entity(name = "carts")
public class Cart {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "cart_id")
    private Long cartId;

    @NotNull
    @Column(name = "order_date")
    private Date orderDate;
}

Reply:
no tak, to jest to co było do tej pory zmerdżowane, to z relacją @ManyToMany jest w pull requescie ktory jeszcze nie jest zaaprobowany

Reply:
> Hmm ale po zrobieniu tych zmian projekt się nie buduje...com.kodilla.ecommercee.EcommerceeApplicationTests > contextLoads FAILED https://prnt.sc/12j1b41 wiesz może co tu jest nie tak?

Wydaje mi się, że tu robiłaś mapowanie na złe pole. Na screenie miałaś tam klasę Product z mapowaniem na cartId, a powinnaś mieć mapowanie na pole przechowujące w klasie Cart listę produktow